account_reconcile_oca/models/account_bank_statement_line.py

```diff
@@ -86,7 +86,11 @@ class AccountBankStatementLine(models.Model):
         store=False,
         default=False,
         prefetch=False,
-        domain=[("rule_type", "=", "writeoff_button")],
+        domain="""
+        [('rule_type', '=', 'writeoff_button'),
+        '|',
+        ('match_journal_ids', '=', False), ('match_journal_ids', 'in', [journal_id])]
```

Comment:
Doesn't this work and it's less confusing?

```suggestion
        ('match_journal_ids', '=', False), ('match_journal_ids', '=', journal_id)]
```